### PR TITLE
Assignment 44 topics and posts

### PR DESCRIPTION
**Pull request by npauzenga**
@Enriikke hey man, would you mind taking a look at this? I'm a little stuck. I initially got a little too ambitious trying to incorporate the `summary` into the `topics#show` and having a "create summary" button appear only when one didn't exist, etc. etc. Turns out I'm having trouble with the basic associations :grin:

If you don't mind taking a look at my routing, I think that's about right and it gives me the routes I'd expect (`/topics/:topic_id/posts/:post_id/summary` and the like), but I'm not seeing the correct summaries on the correct posts.

Also, you'll notice in the `summaries_controller` that I'm not using some of the instance variables I've set up (`@topic`, @post). I have big plans for those but it might make sense to strip those out while I'm trying to get the basics here. Lemme know what you think.

I'm going to press on to the next couple checkpoints in the meantime because I think some of this might be cleared up as we go further.

### CODE REVIEW

File: app/controllers/summaries_controller.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+class SummariesController < ApplicationController
+  def new
+    @summary = Summary.new
```
Comment (Enriikke) [inline app/controllers/summaries_controller.rb]: Since `Summary` is always connected to a `Post`, I think it will be best here to do:

```ruby
@ post = Post.find(params[:id]) # or params[:post_id] depending on your routes ;-)
@summary = @post.build_summary
```

File: app/controllers/summaries_controller.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    authorize @summary
+  end
+
+  def create
+    @topic = Topic.find(params[:topic_id])
+    @post = Post.find(params[:post_id])
+    @summary = Summary.new(summary_params)
```
Comment (Enriikke) [inline app/controllers/summaries_controller.rb]: Same as above, here you can do:

```ruby
@summary = @post.build_summary(summary_params)
```

**Comment by npauzenga** [comment]
@Enriikke so check out these changes. The routing is starting to come together but isn't quite there yet. I've got the summaries showing on `topics#show` (which is where I'd like 'em) but I haven't quite worked out the `summaries#update` routing. There's also more validation to be done (you can break the app pretty easily by looking for summaries that don't exist). Ideally we'd also have the "Edit Summary" button change to "Add Summary" if there isn't one present. That's not a big deal once the rest of it is worked out. I'll keep hacking at this branch and maybe we can discuss in our next meeting.

So basically the original problems I had are resolved (thank you) and it's possible to create and view the correct summary for the correct post. Now it's just about making the summaries an actual, working feature.

File: app/controllers/summaries_controller.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
+class SummariesController < ApplicationController
+  def new
+    @summary = Summary.new
+    authorize @summary
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @summary = Summary.new(summary_params)
+    @post.summary = @summary
```
Comment (Enriikke) [inline app/controllers/summaries_controller.rb]: This is good. I think you can clean up the code a bit by combining both of these statements though:

```ruby
@summary = @post.build_summary(summary_params)
```

**Comment by Enriikke** [comment]
Great work @npauzenga! :shipit:

I added some comments to clean up the code but nothing crucial. There's also a comment of mine from last time about the same thing for the `Summaries#new` action.

Relating showing summaries. Can you tell me what is the error you are getting when you try to look up summaries that don't exist and how are you looking them up?

I think for the `Summaries#update` endpoint, you can decide how you want to do it. I would follow the "standard" Rails-way with `PUT /summaries/{id}`. Based on your `routes.rb` file it seems that you have do to `PUT /posts/{post_id}/summaries/{id}`.

We can go through this checkpoint and talk more about routing during our next session :smile:

**Comment by npauzenga** [comment]
@Enriikke so instead of using a singular resource would you do something like this?

```
resources :posts do
    resources :summary, shallow: true
  end
```

Which would yield:

```
 PUT    /summary/:id
```

**Comment by Enriikke** [comment]
Awesome suggestion @npauzenga!! Yes, I would build my routes using `shallow` most of the time. In cases where you have multiple children resources you can specify it at the parent level:

```ruby
resources :post, shallow: true do
  resources :summary
  resources ...
  ...
end
```

That will make all children `shallow` :smile:.
